pkg/streamingpromql/operators/functions/common.go

```diff
@@ -115,6 +116,10 @@ type FunctionOverRangeVectorDefinition struct {
 
 	// NeedsSeriesNamesForAnnotations indicates that this function uses the names of input series when emitting annotations.
 	NeedsSeriesNamesForAnnotations bool
+
+	// UseFirstArgumentPositionForAnnotations indicates that annotations emitted by this function should use the position of the
+	// first argument, not the position of the inner expression.
+	UseFirstArgumentPositionForAnnotations bool
```

Comment:
what happens if we have more than 1 argument that needs an annotation. Or is this just good enough for now and we'll cross that bridge later.
We're also not allowing for cases of functions with more than 2 arguements. We should add a comment to these limitations.

Reply:
>Or is this just good enough for now and we'll cross that bridge later.

This - this is enough for now, we can revisit it when we have another case we need to deal with.

Reply:
Lets add a comment reasoning about this then. (ie for simplicity given the currently defined promql functions this is sufficient etc but has limitations)

Reply:
Done in 5372b3d